PP-5166 Basic read and write with SQS

kbottla opened the pull request:
## WHAT YOU DID
- Adds a service to send message to SQS and reads messages from SQS

mrlumbu commented on src/main/java/uk/gov/pay/connector/queue/SqsQueueService.java:
```diff
@@ -0,0 +1,44 @@
+        this.sqsClient = sqsClient;
+    }
+
+    public void sendMessage(String queueUrl, String messageBody) {
```
not sure why we're passing the queue url into `sendMessage` - surely that is config that just needs to be injected?

mrlumbu commented on src/main/java/uk/gov/pay/connector/queue/SqsQueueReceiveResponse.java:
```diff
@@ -0,0 +1,41 @@
+    private String receiptHandle;
+    private String messageBody;
+
+    private SqsQueueReceiveResponse(String messageId, String receiptHandle, String messageBody) {
```
this name wasn't clear to me on first reading - response made me think it was an http thing, but really it's just a pojo to wrap a message right? In fact I'm not sure what we're gaining here, as we're just swapping one Sqs specific class for another (`sqs.model.Message` for `SqsQueueReceiveResponse`)

kbottla replied:
Agree! after a bit of thinking renamed it to `QueueMessage`. If you have any other suggestions happy to take them into account.

It is almost a swap with SQS specific class so to avoid passing SQS related objects (`ReceiveMessageResult`) throughout the code base.

mrlumbu commented on src/test/java/uk/gov/pay/connector/queue/SqsQueueServiceTest.java:
```diff
@@ -0,0 +1,106 @@
+@RunWith(MockitoJUnitRunner.class)
+public class SqsQueueServiceTest {
+
+    private static final String CAPTURE_QUEUE_URL = "http://capture-url";
```
the implementation of the queue service is totally agnostic of what it is for, so the test should probably also be generic - i.e it shouldn't be talking about specific things like capture etc

kbottla commented:
Thanks @mrlumbu . I have addressed your comments

>  I'm also interested in how you intend to expose this. i.e. Nothing else should know it's interacting with an sqs queue.

I envisage having the below structure when it is completed

**Writing to the queue**
`CardCaptureService -> CaptureQueue.sendForCapture(charge) -> SqsQueueService.sendMessage()`

**Reading from the queue**
`CardCaptureProcess -> CaptureQueue.getChargesToCapture()  gets List<QueueMessage> and converts them as required by CardCaptureProcess -> SqsQueueService.receiveMessages() returns CardCaptureProcess`

Happy to discuss them further..
